File: pinyougou-interface/src/main/java/cn/itcast/core/service/AddressService.java
package cn.itcast.core.service;

import cn.itcast.core.pojo.address.Address;

import java.util.List;

public interface AddressService {
    List<Address> findListByLoginUser(String name);


    void addAddress(Address address);

    void delete(Long id);

    void setDefault(Long id, String name);

    Address findOneById(Long id);

    void update(Address address);
}

File: pinyougou-web-user/src/main/webapp/js/service/addressService.js
// 定义服务层:
app.service("addressService",function($http){
	this.findListByLoginUser = function(){
		return $http.get("../address/findListByLoginUser.do");
	}

	//新增地址
	this.add=function(entity){
        return $http.post('../address/addAddress.do?',entity);
    }

    //修改地址
    this.update=function(entity){
        return $http.post('../address/update.do?',entity);
    }
    //删除地址
    this.dele = function(id){
        return $http.get("../address/delete.do?id="+id);
    }
	//设为默认地址
    this.setDefault = function(id){
        return $http.get("../address/setDefault.do?id="+id);
    }

    this.findOneById=function(id){
        return $http.get("../address/findOneById.do?id="+id);
    }

});